data/data/baremetal/masters/outputs.tf

```
@@ -0,0 +1,11 @@
+output "control_plane_ips" {
+  value = <<EOT
+    var.master_count > 2 ? [
+      data.ironic_introspection.openshift-master-introspection.interfaces.*.ip ] :
+        var.master_count > 1 ? [
+          data.ironic_introspection.openshift-master-introspection.interfaces.0.ip,
+          data.ironic_introspection.openshift-master-introspection.interfaces.1.ip ] :
+            var.master_count > 0 ? [
+              data.ironic_introspection.openshift-master-introspection.interfaces.0.ip ] : 0
+EOT
```

This does not seem quite right to me. I would expect the master count to effect the field after `openshift-master-inspection`, as that is a list of masters. You are using the count to select the network interface in a given master. I would expect something like the following instead to get the IPs for all of the interfaces on all of the masters. Even with that, though, we are missing some logic for selecting the IP in the machine network.
```suggestion
  value = [ data.ironic_introspection.openshift-master-introspection.*.interfaces.*.ip ]
```